### ISSUE TITLE
[Plugin manager] Small improvement?

### ISSUE THREAD

**Issue by rubyFeedback** [geany/geany#2142]
When clicking on "Plugin Manager", a gtk label (I assume) appears:

"Choose which plugins should be loaded at startup:"

Would it be possible to also display the current version of the geany-plugins in that widget somewhere? I upgraded from 1.34 and was not sure whether I compiled geany-plugins yet and whether it could be found (since it would compile into a non-standard path); that way I could quickly see which version is in use.

**Comment by elextr** [comment]
The tooltip that comes up when you hover over a line on the plugin manager list shows the version of the plugin.

**Comment by LarsGit223** [comment]
I do not agree that this is an existing capability. Yes, the tooltip shows the version of the plugin, e.g. version 1.32 for the TreeBrowser plugin. But it does not tell you from which package of geany plugins it is loaded.

Also, I disagree to place the version of geany-plugins in the plugin manager. I suggest to place it in the about dialog.

**Comment by elextr** [comment]
> But it does not tell you from which package of geany plugins it is loaded.

Ok, to expand, geany-plugins is only a collection for installation convenience, its not an actual plugin, so there is no information about it available to Geany.  And many distros provide plugins as individual packages which can be installed individually and "geany-plugins" is nothing more than a meta package that references the others, it doesn't install anything outside the package database.

Also plugins do not just come from geany-plugins, there are "third party" plugins listed on [Geany's website](https://plugins.geany.org/) and others in places like Sourceforge and Github.  

So the only version that makes sense is the version of the individual plugin as is currently shown.

**Comment by elextr** [comment]
From my Synaptic for geany-plugins "This is a metapackage which installs the whole set of plugins for Geany." and its file list:

```
/usr/share/doc/geany-plugins/changelog.Debian.gz
/usr/share/doc/geany-plugins/copyright
```

**Comment by LarsGit223** [comment]
> So the only version that makes sense is the version of the individual plugin as is currently shown.

I think it makes sense to see the package also as no one will ever take a plugin out of the package and install it stand-alone. Looking at the plugin API I saw that the version is a string. So plugin developers could theoretically add the PACKAGE_STRING to the version. But that would meany each plugin of a package would need to do this manually to have a consistent look, so not very beautiful.

Anyway, I would rate this just an inconvenience.

**Comment by b4n** [comment]
As @elextr said, there isn't really anything such as "geany-plugins" when it comes to Geany or the plugin manager -- geany-plugins is just an umbrella for a collection of individual plugins.

> I think it makes sense to see the package also as no one will ever take a plugin out of the package and install it stand-alone.

Take it out, no, but it's totally possible to install only some of the plugins, and they all are disabled automatically if their dependencies are missing at build time and they didn't get explicitly enabled -- meaning it's extremely easy to get a partial collection.
And as @elextr mentioned, when using downstream distributors, it's even more common to have each plugin separate, and possibly a meta-package merely installing them all.

> Looking at the plugin API I saw that the version is a string. So plugin developers could theoretically add the PACKAGE_STRING to the version. But that would meany each plugin of a package would need to do this manually to have a consistent look, so not very beautiful.

Yes, each plugin could set its version to include the "geany-plugins" version in it.  Or it could simply use the "geany-plugins" version as the plugin's version, which quite a few plugins do.  **If** there is enough of a rationale, I could imagine all plugins adopting this versioning scheme.

---

All this said, if it's really to make sure the expected build that is used, better look at the filename for the plugin library that is displayed in the tooltip when hovering.

**Comment by elextr** [comment]
> So plugin developers could theoretically add the PACKAGE_STRING to the version.

@LarsGit223 Thats up to the plugin developers, and given some plugins are unmaintained its unlikely that it will be kept up to date, so it becomes another maintenance chore for the releasers.

> All this said, if it's really to make sure the expected build that is used, better look at the filename for the plugin library that is displayed in the tooltip when hovering.

@b4n That tells you the SO filename and location, but how does it tell you the version?

**Comment by b4n** [comment]
> @b4n That tells you the SO filename and location, but how does it tell you the version?

There's the version as well, but I think the SO filename would be of interest given the OP states:

> […] I upgraded from 1.34 and was not sure whether I compiled geany-plugins yet and whether it could be found (since it would compile into a non-standard path); […]

**Comment by elextr** [comment]
> […] I upgraded from 1.34 and was not sure whether I compiled geany-plugins yet and whether it could be found (since it would compile into a non-standard path); […]

Well, the plugins simply don't show in the dialog if they were not built into the same prefix.
